includes/Checker/Checks.php

```diff
@@ -0,0 +1,123 @@
+<?php
+/**
+ * Class WordPress\Plugin_Check\Checker\Checks
+ *
+ * @package plugin-check
+ */
+
+namespace WordPress\Plugin_Check\Checker;
+
+use WordPress\Plugin_Check\Checker\Check_Context;
+use Exception;
+
+/**
+ * Class to run checks on a plugin.
+ *
+ * @since n.e.x.t
+ */
+class Checks {
+
+	/**
+	 * Context for the plugin to check.
+	 *
+	 * @since n.e.x.t
+	 * @var Check_Context
+	 */
+	protected $check_context;
+
+	/**
+	 * Sets the main context and the main file of the plugin to check.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $plugin_main_file Absolute path to the plugin main file.
+	 */
+	public function __construct( $plugin_main_file ) {
+		$this->check_context = new Check_Context( $plugin_main_file );
+	}
+
+	/**
+	 * Runs checks against the plugin.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $checks An array of Check objects to run.
+	 * @return Check_Result Object containing all check results.
+	 *
+	 * @throws Exception Thrown when check fails with critical error.
+	 */
+	public function run_checks( array $checks ) {
+		$result     = new Check_Result( $this->check_context );
```

Comment:
@jjgrainger Do we needs to add `use WordPress\Plugin_Check\Checker\Check_Result;`?

Reply:
@mukeshpanchal27 it's not necessarily a requirement as it shares the same namespace, but I see no harm in explicitly defining this.

Reply:
Not really needed, I think we can avoid that in the future. But okay to keep for now.